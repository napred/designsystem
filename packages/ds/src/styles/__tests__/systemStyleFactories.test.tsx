/**
 * @jest-environment jsdom
 */

import React from 'react';
// tslint:disable-next-line:no-implicit-dependencies
import { render } from 'react-testing-library';

import { createComponent, DesignSystem } from '../../';
import {
  createNumericSystemStyle,
  createStringSystemStyle,
  createSystemStyle,
} from '../systemStyleFactories';

const Box = createComponent('Box', 'div');

describe('styler factories', () => {
  describe('createSystemStyle', () => {
    it('works with simple value (without default value)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val)];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with simple value (with default value)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val, 1)];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box fontSize={0} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (without default value)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val)];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={['1', '2']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={1} styles={styles}>
          <Box test={['1', '2']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default simple value)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val, '2')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[0]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default responsive value)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val, ['1', '2'])];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={['1']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes the value from default responsive value if is not specified in prop (with default)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val, ['1', '2'])];

      const { asFragment, rerender } = render(
        <DesignSystem is={1} styles={styles}>
          <Box fontSize={['1']} />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={1} styles={styles}>
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes last value from responsive value if breakpoint exceeds the range (with default responsive)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val, ['1', '2'])];

      const { asFragment, rerender } = render(
        <DesignSystem is={1} styles={styles}>
          <Box test={['1']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={['1']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={['1', '3']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes last value from responsive value if breakpoint exceeds the range (with default simple)', () => {
      const styles = [createSystemStyle('test', 'test', 'spacing', val => val, '1')];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={['1']} />
          <Box test={['1', '3']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={['1']} />
          <Box test={['1', '3']} />
          <Box />,
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('createSystemNumericStyler', () => {
    it('works with simple value (without default value)', () => {
      const styles = [createNumericSystemStyle('test', 'test', 'fontSizes')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with simple value (with default value)', () => {
      const styles = [createNumericSystemStyle('test', 'test', 'fontSizes', 1)];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (without default value)', () => {
      const styles = [createNumericSystemStyle('test', 'test', 'fontSizes')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={['1', '2']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default simple value)', () => {
      const styles = [createNumericSystemStyle('test', 'test', 'fontSizes', '2')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[0]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default responsive value)', () => {
      const styles = [createNumericSystemStyle('test', 'test', 'fontSizes', ['1', '2'])];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={['1']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('createSystemStringStyler', () => {
    it('works with simple value (without default value)', () => {
      const styles = [createStringSystemStyle('test', 'test', 'colors')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test="white" />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with simple value (with default value)', () => {
      const styles = [createStringSystemStyle('test', 'test', 'colors', 'white')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test="black" />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (without default value)', () => {
      const styles = [createStringSystemStyle('test', 'test', 'colors', 'white')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={['white', 'black']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default simple value)', () => {
      const styles = [createStringSystemStyle('test', 'test', 'colors', 'red')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={['white', 'red']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default responsive value)', () => {
      const styles = [createStringSystemStyle('test', 'test', 'colors', ['white', 'red'])];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={['white']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={1} styles={styles}>
          <Box test={['white']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
