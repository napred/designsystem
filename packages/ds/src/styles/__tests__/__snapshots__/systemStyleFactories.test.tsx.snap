// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styler factories createSystemNumericStyler works with responsive value (with default responsive value) 1`] = `
<DocumentFragment>
  .css-1pqr7g2 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 20px;
}

<div
    class="css-1pqr7g2"
  />
  .css-1pqr7g2 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 20px;
}

<div
    class="css-1pqr7g2"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemNumericStyler works with responsive value (with default simple value) 1`] = `
<DocumentFragment>
  .css-1mdabbm {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 16px;
}

<div
    class="css-1mdabbm"
  />
  .css-jcrk0j {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 24px;
}

<div
    class="css-jcrk0j"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemNumericStyler works with responsive value (without default value) 1`] = `
<DocumentFragment>
  .css-1pqr7g2 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 20px;
}

<div
    class="css-1pqr7g2"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemNumericStyler works with simple value (with default value) 1`] = `
<DocumentFragment>
  .css-1pqr7g2 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 20px;
}

<div
    class="css-1pqr7g2"
  />
  .css-1pqr7g2 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 20px;
}

<div
    class="css-1pqr7g2"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemNumericStyler works with simple value (without default value) 1`] = `
<DocumentFragment>
  .css-1pqr7g2 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 20px;
}

<div
    class="css-1pqr7g2"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStringStyler works with responsive value (with default responsive value) 1`] = `
<DocumentFragment>
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStringStyler works with responsive value (with default responsive value) 2`] = `
<DocumentFragment>
  .css-9sndi4 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #EA2E49;
}

<div
    class="css-9sndi4"
  />
  .css-9sndi4 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #EA2E49;
}

<div
    class="css-9sndi4"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStringStyler works with responsive value (with default simple value) 1`] = `
<DocumentFragment>
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
  .css-9sndi4 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #EA2E49;
}

<div
    class="css-9sndi4"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStringStyler works with responsive value (without default value) 1`] = `
<DocumentFragment>
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStringStyler works with simple value (with default value) 1`] = `
<DocumentFragment>
  .css-109oslm {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: black;
}

<div
    class="css-109oslm"
  />
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStringStyler works with simple value (without default value) 1`] = `
<DocumentFragment>
  .css-g4jko9 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: #FFFFFF;
}

<div
    class="css-g4jko9"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes last value from responsive value if breakpoint exceeds the range (with default responsive) 1`] = `
<DocumentFragment>
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes last value from responsive value if breakpoint exceeds the range (with default responsive) 2`] = `
<DocumentFragment>
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes last value from responsive value if breakpoint exceeds the range (with default responsive) 3`] = `
<DocumentFragment>
  .css-1mdabbm {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 16px;
}

<div
    class="css-1mdabbm"
  />
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes last value from responsive value if breakpoint exceeds the range (with default simple) 1`] = `
<DocumentFragment>
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes last value from responsive value if breakpoint exceeds the range (with default simple) 2`] = `
<DocumentFragment>
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-1mdabbm {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 16px;
}

<div
    class="css-1mdabbm"
  />
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  ,
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes the value from default responsive value if is not specified in prop (with default) 1`] = `
<DocumentFragment>
  .css-1eksjgu {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 20px;
  test: 8px;
}

<div
    class="css-1eksjgu"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle takes the value from default responsive value if is not specified in prop (with default) 2`] = `
<DocumentFragment>
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle works with responsive value (with default responsive value) 1`] = `
<DocumentFragment>
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle works with responsive value (with default simple value) 1`] = `
<DocumentFragment>
  .css-1yk49p6 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 0;
}

<div
    class="css-1yk49p6"
  />
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle works with responsive value (without default value) 1`] = `
<DocumentFragment>
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle works with responsive value (without default value) 2`] = `
<DocumentFragment>
  .css-ch3tt0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 8px;
}

<div
    class="css-ch3tt0"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle works with simple value (with default value) 1`] = `
<DocumentFragment>
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
</DocumentFragment>
`;

exports[`styler factories createSystemStyle works with simple value (without default value) 1`] = `
<DocumentFragment>
  .css-cvk9v3 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 4px;
}

<div
    class="css-cvk9v3"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;
